src/typescript/frontend/src/components/charts/PrivateChart.tsx

```diff
@@ -145,119 +145,117 @@ export const Chart = async (props: ChartContainerProps) => {
 
         setTimeout(() => onSymbolResolvedCallback(symbolInfo), 0);
       },
-      getBars: async (symbolInfo, resolution, periodParams, onHistoryCallback, onErrorCallback) => {
+      getBars: async (
+        _symbolInfo,
+        resolution,
+        periodParams,
+        onHistoryCallback,
+        _onErrorCallback
+      ) => {
         const { from, to } = periodParams;
-        try {
-          const period = ResolutionStringToPeriod[resolution.toString()];
-          const periodDuration = periodEnumToRawDuration(period);
-          const start = new Date(from * 1000);
-          const end = new Date(to * 1000);
-          // TODO: Consider that if our data is internally consistent and we run into performance/scalability issues
-          // with this implementation below (fetching without regard for anything in state), we can store the values in
-          // state and coalesce that with the data we fetch from the server.
-          const data = await fetchAllCandlesticksInTimeRange({
-            marketID: props.marketID,
-            start,
-            end,
-            period,
-          });
+        const period = ResolutionStringToPeriod[resolution.toString()];
+        const periodDuration = periodEnumToRawDuration(period);
+        const periodDurationSeconds = (periodDuration / PeriodDuration.PERIOD_1M) * 60;
+        const end = new Date(to * 1000);
+        // The start timestamp is rounded so that all the people who load the webpage at a similar time get served
+        // the same cached response.
+        const start = from - (from % periodDurationSeconds);
+        const data: PeriodicStateEventModel[] = await fetch(
+          `/candlesticks?marketID=${props.marketID}&start=${start}&period=${period}&limit=500`
+        )
```

Comment:
I think you should use the `searchParams` fetch field here, it's difficult to read/parse a raw URL string. There should be one where you can specify the query params as `key: value`s